src/dwn/interfaces/records.js

```diff
@@ -0,0 +1,103 @@
+import { DwnConstant } from '@tbd54566975/dwn-sdk-js';
+
+import { Interface } from './interface.js';
+import { Record } from '../models/record.js';
+import { dataToBytes } from '../../utils.js';
+
+export class Records extends Interface {
+  constructor(dwn) {
+    super(dwn, 'Records');
+  }
+
+  get create() {
+    return this.write;
+  }
+
+  async createFrom(target, request) {
+    const { author: inheritedAuthor, target: _, ...inheritedProperties } = request.record.toJSON();
+
+    // If `data` is being updated then `dataCid` and `dataSize` must not be present.
+    if (request?.data !== undefined) {
+      delete inheritedProperties.dataCid;
+      delete inheritedProperties.dataSize;
+    }
+
+    // If `published` is set to false, ensure that `datePublished` is undefined. Otherwise, DWN SDK's schema validation
+    // will throw an error if `published` is false but `datePublished` is set.
+    if (request?.message?.published === false && inheritedProperties?.datePublished !== undefined) {
+      delete inheritedProperties.datePublished;
+      delete inheritedProperties.published;
+    }
+
+    // If the request changes the `author` or message `descriptor` then the deterministic `recordId` will change.
+    // As a result, we will discard the `recordId` if either of these changes occur.
+    if ((request?.message && Object.keys(request.message).length > 0)
```

Comment:
can we use `!isEmptyObject(request.message)` instead?

Reply:
Addressed in 96c7a8030cff76db65305578bc3beb34ffc6fc51